src/transformStateWithClones.js

```diff
@@ -7,7 +7,37 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const result = [];
+  let currentState = { ...state };
+
+  for (const action of actions) {
+    switch (action.type) {
+      case 'addProperties':
+        const { extraData } = action;
+
+        currentState = { ...currentState, ...extraData };
+        break;
+
+      case 'removeProperties':
+        const { keysToRemove } = action;
+
+        for (const key of keysToRemove) {
+          if (currentState.hasOwnProperty(key)) {
```

Comment:
The `hasOwnProperty` check is unnecessary because the `delete` operator will not throw an error if the property does not exist. You can safely remove this check to simplify the code.